framework/src/util/Utils.cpp

```diff
@@ -154,6 +154,13 @@ std::string GetCurrentWorkingDirectory()
   return std::string();
 }
 
+static const std::string s_CurrWorkingDir = InitCurrentWorkingDirectory();
```

Comment:
global statics are problematic. This doesn't look like it will cause any static destruction issues now. As the code evolves, it may in future. This is more of a future looking code maintenance concern than anything else.
Can a function local static be used instead? That is thread-safe and guaranteed to be called once (for C++11).